modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/KMPSearch.kt

```diff
@@ -0,0 +1,74 @@
+package au.com.shiftyjelly.pocketcasts.player.view.transcripts
+
+import javax.inject.Inject
+
+// An implementation of the Knuth-Morris-Pratt algorithm
+class KMPSearch @Inject constructor() {
```

Comment:
This reuses [iOS search algorithm](https://github.com/Automattic/pocket-casts-ios/pull/1982/files#diff-e41c664a747a2b673002d0b507fe6f3f438c8ae84fd09c3cae591154f6e0662d).